### PR TITLE
modularized flavor info

### PR DESCRIPTION
**Pull request by fme-samuelbirk**

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,25 @@
+0 info it worked if it ends with ok
```
Comment (FME-nstokoe) [inline npm-debug.log]: shouldn't this file be ignored?

File: modules/get_flavor_info.coffee [hunk 2]
```diff
@@ -0,0 +1,11 @@
+plugins  = require('gulp-load-plugins')()
+flavors   = require('../config').flavors
+urban_airship_default   = require('../config').urban_airship_default
+argv     = require('yargs').argv
+
```
Comment (FME-nstokoe) [inline modules/get_flavor_info.coffee]: getFlavorInfo = ->name: if argv.flavor then argv.flavor else 'development'
+  directory: if argv.flavor then flavors[argv.flavor].institution_directory else flavors.development.institution_directory
+  package: if argv.flavor then "com.fullmeasureed.#{argv.flavor.replace(/_/g, '')}" else 'com.fullmeasureed.fmedev'
+  urban_airship: if argv.flavor && flavors[argv.flavor].urban_airship then flavors[argv.flavor].urban_airship else urban_airship_default
+module.exports = getFlavorInfo

Comment (FME-nstokoe) [inline modules/get_flavor_info.coffee]: How about define a <code>default_flavor</code> object that can be referenced below so it only has be changed in one place if we rename it

File: tasks/release/customize-config.coffee [hunk 3]
```diff
@@ -1,17 +1,11 @@
...
 gulp.task 'customize-config', ->
-  package_name = if argv.flavor then "com.fullmeasureed.#{argv.flavor.replace(/_/g, '')}" else 'com.fullmeasureed.fmedev'
-  flavor_name = if argv.flavor then flavors[argv.flavor].name.replace(/_/g, '') else flavors.development.name
-  if argv.flavor && flavors[argv.flavor].urban_airship
-    urban_airship = flavors[argv.flavor].urban_airship
-  else
-    urban_airship = urban_airship_default
+  package_name = getFlavorInfo().package
+  flavor_name = getFlavorInfo().name
+  urban_airship = getFlavorInfo().urban_airship
```
Comment (FME-nstokoe) [inline tasks/release/customize-config.coffee]: doesn't look like this is used in the sed command

Comment (fme-samuelbirk) [inline tasks/release/customize-config.coffee]: It is but you have to expand the code below to see it

**Pull request merged**